lib/chef/provider/mount/linux.rb

```diff
@@ -0,0 +1,60 @@
+#
+# Author:: Antima Gupta (<agupta@chef.io>)
+# Copyright:: Copyright (c) Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../mount"
+
+class Chef
+  class Provider
+    class Mount
+      class Linux < Chef::Provider::Mount::Mount
+
+        provides :mount, os: "linux"
+
+        # Check to see if the volume is mounted.
+        # "findmnt" outputs the mount points with volume.
+        # Convert the mount_point of the resource to a real path in case it
+        # contains symlinks in its parents dirs.
+
+        def mounted?
+          mounted = false
+
+          real_mount_point = if ::File.exists? @new_resource.mount_point
+                               ::File.realpath(@new_resource.mount_point)
+                             else
+                               @new_resource.mount_point
+                             end
+
+          shell_out!("findmnt -rn").stdout.each_line do |line|
+            case line
+            when /\A#{Regexp.escape(real_mount_point)}\s+#{device_mount_regex}\s/
```

Comment:
This looks really nice. One thing that would be useful though would be to add a comment for each `when` that has an example line you're matching.

Reply:
Permalinks to rubular are super handy in this sort of case https://rubular.com/

Reply:
Agreed, I have added Permalink.